pypredict/trades.py

```diff
@@ -156,10 +151,11 @@ async def subscribe(self):
         topic_id = await self.ensign.topic_id(self.sub_topic)
 
         # Subscribe to the topic.
-        # TODO: Handle dropped stream, but the SDK should really handle this.
-        async for event in self.ensign.subscribe(topic_id):
-            data = json.loads(event.data)
-            await self.run_model_pipeline(data)
+        # self.run_model_pipeline is a callback function that gets executed when 
+        # a new event arrives in the topic
+        await self.ensign.subscribe(topic_id, on_event=self.run_model_pipeline)
+        # awaiting the Future means the coroutine will wait for the callback function to execute
+        await asyncio.Future()
```

Comment:
Technically I think what this is doing is creating a Future and awaiting its result, since nothing in the code is setting the result of the Future then it waits forever.

Reply:
that is a much more clearer explanation - I have updated the comment.